network/p2p/topic_validator_test.go

```diff
@@ -0,0 +1,122 @@
+package p2p
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/onflow/flow-go/crypto"
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/utils/unittest"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTopicValidator(t *testing.T) {
+	identity1, privateKey1 := createID(t)
+	node1 := createNode(t, identity1.NodeID, privateKey1)
+
+	identity2, privateKey2 := createID(t)
+	node2 := createNode(t, identity2.NodeID, privateKey2)
+
+	badTopic := engine.TopicFromChannel(engine.SyncCommittee, rootBlockID)
+
+	validator := StakedValidator{func() flow.IdentityList {
+		return flow.IdentityList{identity1, identity2}
+	}}
+
+	unstakedKey, err := unittest.NetworkingKey()
```

Comment:
Should we need a `unittest.UnstakedNetworkingKey()` at some point? happy to add one if you open an issue.

Reply:
So currently I think we are okay? And eventually once we no longer need a mapping from Flow IDs to Peer IDs for unstaked nodes (because unstaked nodes will not need a Flow ID), then we will also not need to constrain unstaked networking keys to a specific format, and hence we will not need a `unittest.UnstakedNetworkingKey()`.

Also, I think if we do need something like `unittest.UnstakedNetworkingKey()`, I could actually just use the unstaked key generation function which you've already implemented :)